### PR TITLE
Feat/core 2

### PR DESCRIPTION
**Pull request by laiheau**

### CODE REVIEW

File: source/client/client.py
Comment (MatthiasvonRakowski) [inline source/client/client.py:1]: You have to put everything in a function or class

File: source/client/memory.py
Comment (ramosleandre) [inline source/client/memory.py:1]: You need to also put the code in a function or class and create a main function

File: source/server/core.py
Comment (MatthiasvonRakowski) [inline source/server/core.py:1]: You have to put your global variable in a @ dataclass

File: source/client/client.py
Comment (MatthiasvonRakowski) [reply source/client/client.py:1]: You have to put your global variable in a @ dataclass

File: source/server/server.py
Comment (MatthiasvonRakowski) [inline source/server/server.py:1]: Put your global variable in a dataclass and you have to put any line in a class or function

Comment (MatthiasvonRakowski) [reply source/server/server.py:1]: server must have a class

File: source/client/client.py
Comment (MatthiasvonRakowski) [reply source/client/client.py:1]: client must have a class

File: source/server/core.py
Comment (MatthiasvonRakowski) [reply source/server/core.py:1]: core must be a class

**Review by MatthiasvonRakowski — CHANGES_REQUESTED** [review]
Check comments

**Review by ramosleandre — CHANGES_REQUESTED** [review]
check the comments :)

**Review by pierrelissope — APPROVED** [review]

**Review by LixiosDelios — APPROVED** [review]

**Pull request merged**